### PR TITLE
Feat/new app frontend

### PR DESCRIPTION
**Pull request by Datayama38**
## Release Notes

- **New Features**
	- Introduced multiple new pages including **Home**, **Administration**, **Activite**, and **Proconnect** for improved navigation and user experience.
	- Added a **Search** configuration for enhanced API interactions.
	- Implemented a **Footer** and **Header** for consistent site navigation.
	- Added a **security policy** document to improve user trust and transparency.
	- Introduced a new **AuthButton** for streamlined authentication processes.
	- Added **Analytics** tracking for better user insights.

- **Documentation**
	- Created a comprehensive **README.md** for project setup and usage instructions.
	- Added **security.txt** and **security-policy.txt** for security vulnerability reporting.

- **Chores**
	- Introduced configuration files for environment variables and ESLint settings to streamline development.
	- Added a **gitignore** file to exclude unnecessary files from version control. 

- **Style**
	- Implemented global styles including a **text-center** class for better text alignment. 

- **Enhancements**
	- Updated TypeScript and ESLint configurations for improved code quality and maintainability.
	- Enhanced authentication context for better state management and error handling.
	- Improved URL parameter handling and nonce generation for secure authentication processes.

### CODE REVIEW

File: app-front/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "prebuild": "only-include-used-icons"
+  },
+  "dependencies": {
+    "@codegouvfr/react-dsfr": "^1.14.5",
+    "@emotion/react": "^11.13.3",
+    "@emotion/styled": "^11.13.0",
+    "@mui/material": "^6.1.6",
+    "@socialgouv/matomo-next": "^1.9.2",
+    "keycloak-js": "^26.0.5",
+    "next": "14.2.17",
```
Comment (P3rceval) [inline app-front/package.json:20]: On peut peut-être passer en version 15 ?

File: app-front/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "predev": "only-include-used-icons",
+    "prebuild": "only-include-used-icons"
+  },
+  "dependencies": {
+    "@codegouvfr/react-dsfr": "^1.14.5",
+    "@emotion/react": "^11.13.3",
+    "@emotion/styled": "^11.13.0",
+    "@mui/material": "^6.1.6",
+    "@socialgouv/matomo-next": "^1.9.2",
+    "keycloak-js": "^26.0.5",
```
Comment (P3rceval) [inline app-front/package.json:19]: Keycloak-js à spprimer du coup ?

**Review by P3rceval — CHANGES_REQUESTED** [review]

File: app-front/package.json (hunk 2 above)
Comment (Datayama38) [reply app-front/package.json:19]: oui c'est à supprimer

File: app-front/package.json (hunk 1 above)
Comment (Datayama38) [reply app-front/package.json]: done

**Pull request merged**